src/lib/shady-render.ts

```diff
@@ -162,8 +162,18 @@ const prepareTemplateStyles =
       if (window.ShadyCSS!.nativeShadow) {
         // When in native Shadow DOM, re-add styling to rendered content using
         // the style ShadyCSS produced.
-        const style = template.element.content.querySelector('style')!;
-        renderedDOM.insertBefore(style.cloneNode(true), renderedDOM.firstChild);
+        const style = template.element.content.querySelector('style');
+        // Note, an empty style will be eliminated so don't try to re-insert it.
+        if (style !== null) {
+          renderedDOM.insertBefore(
+              style.cloneNode(true), renderedDOM.firstChild);
+          // However, we do need to ensure the template has a style in it if
+          // ShadyCSS eliminated it.
+        } else {
+          template.element.content.insertBefore(
```

Comment:
This seems like the right spot for the comment about Shady CSS and empty style nodes

Reply:
Updated